BraveWallet/Crypto/Stores/NetworkStore.swift

```diff
@@ -12,54 +12,77 @@ import Strings
 ///
 /// This wraps a JsonRpcService that you would obtain through BraveCore and makes it observable
 public class NetworkStore: ObservableObject {
-  @Published private(set) var ethereumChains: [BraveWallet.NetworkInfo] = []
-  @Published private(set) var selectedChainId: String = BraveWallet.MainnetChainId
-
-  var selectedChain: BraveWallet.NetworkInfo {
-    ethereumChains.first(where: { $0.chainId == self.selectedChainId }) ?? .init()
+  
+  enum SetSelectedChainError: Error {
+    case selectedChainHasNoAccounts
+    case chainAlreadySelected
+    case unknown
   }
+  
+  @Published private(set) var allChains: [BraveWallet.NetworkInfo] = []
 
-  var selectedChainBinding: Binding<BraveWallet.NetworkInfo> {
-    .init(
-      get: { self.ethereumChains.first(where: { $0.chainId == self.selectedChainId }) ?? .init() },
-      set: {
-        self.setSelectedChain(chainId: $0.chainId, coin: $0.coin)
-      }
-    )
+  @Published private(set) var selectedChainId: String = BraveWallet.MainnetChainId
+  var selectedChain: BraveWallet.NetworkInfo {
+    allChains.first(where: { $0.chainId == self.selectedChainId }) ?? .init()
   }
 
+  private let keyringService: BraveWalletKeyringService
   private let rpcService: BraveWalletJsonRpcService
   private let walletService: BraveWalletBraveWalletService
 
   public init(
+    keyringService: BraveWalletKeyringService,
     rpcService: BraveWalletJsonRpcService,
     walletService: BraveWalletBraveWalletService
   ) {
+    self.keyringService = keyringService
     self.rpcService = rpcService
     self.walletService = walletService
     self.updateChainList()
-    walletService.selectedCoin { [self] coin in
-      rpcService.chainId(coin) { chainId in
-        self.selectedChainId = chainId.isEmpty ? BraveWallet.MainnetChainId : chainId
-      }
-    }
     rpcService.add(self)
+    
+    Task { @MainActor in // fetch current selected network
+      let selectedCoin = await walletService.selectedCoin()
+      let chainIdForCoin = await rpcService.chainId(selectedCoin)
+      let allNetworksForCoin = await rpcService.allNetworks(selectedCoin)
+      guard let chain = allNetworksForCoin.first(where: { $0.chainId == chainIdForCoin }) else { return }
```

Comment:
is this the same as 
let chain = rpcService.network(selectedCoin)
?

Reply:
Nice catch, yes I think so. Will double check.

Reply:
[fa9183e](https://github.com/brave/brave-ios/pull/5681/commits/fa9183e24964c4e1c06e2382efbe259a852f7f8c)